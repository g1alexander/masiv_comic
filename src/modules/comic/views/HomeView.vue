<script lang="ts" setup>
import { useComic } from '../composables'

const { loader, comic, setStar, starClass } = useComic()
</script>

<template>
  <main class="home">
    <template v-if="comic.getIsLoading">
      <img :src="loader" alt="loading..." width="180" />
    </template>

    <section class="home-section" v-else>
      <h1 class="home-section__h1">{{ comic.getComic.title.toUpperCase() }}</h1>
      <img class="home-section__img" :src="comic.getComic.img" :alt="comic.getComic.alt" />

      <div class="home-section__stars">
        <h3>Puntuacion:</h3>
        <p>
          <template v-for="star in 5" :key="star">
            <i
              @click="setStar(star)"
              class="home-section__stars__icon"
              :class="starClass(star)"
            ></i>
          </template>
        </p>
      </div>

      <article class="home-section__btns">
        <button v-if="comic.getPage > 1" @click="comic.setPage(comic.getPage - 1)">prev</button>
        <button @click="comic.setPage(comic.getPage + 1)">next</button>
      </article>
    </section>
  </main>
</template>
