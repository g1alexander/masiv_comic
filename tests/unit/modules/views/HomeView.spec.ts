import { describe, expect, test } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import HomeView from '@/modules/comic/views/HomeView.vue'

describe('component ExtraFormView.vue', () => {
  const wrapper = shallowMount(HomeView)

  test('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
