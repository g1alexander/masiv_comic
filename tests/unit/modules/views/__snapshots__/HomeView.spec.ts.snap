// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component ExtraFormView.vue > to match snapshot 1`] = `
"<main>
  <h1>hola</h1>
</main>"
`;
